### ISSUE TITLE
Problems setting up the demo

### ISSUE THREAD

**Issue by Jayasheel** [spiffe/spire-controller-manager#1]
ERROR   error received after stop sequence was engaged  {"error": "context canceled"}
E0303 18:08:57.569279     977 leaderelection.go:330] error retrieving resource lock spire-system/98c9c988.spiffe.io: Get "https://10.96.0.1:443/api/v1/namespaces/spire-system/configmaps/98c9c988.spiffe.io": context canceled
ERROR   setup   problem running manager {"error": "open /tmp/k8s-webhook-server/serving-certs/tls.crt: no such file or directory"}

Getting this error in the "spire-controller-manager" container logs while bringing up spire-server pod. Have followed the demo bring up instructions as documented. Am I missing anything?

**Comment by azdagron** [comment]
Looks like the controller manager was waiting to obtain the leader election lock? Was this happening in both clusters or just one? When you applied the config for each cluster, did you use the correct `cluster1` or `cluster2` script to make sure kubectl was targeting the correct cluster?

**Comment by chris13524** [comment]
I am running the demo as instructed by the README, but I'm experiencing this same issue:

```
$ ./cluster1 kubectl -n spire-system logs -f spire-server-545f657995-k4c6l spire-controller-manager
2022-04-13T01:51:17.983Z        INFO    setup   Config loaded   {"cluster name": "cluster1", "trust domain": "cluster1.demo", "ignore namespaces": ["kube-system", "kube-public", "spire-system", "metallb-system", "local-path-storage"], "gc interval": "10s", "spire socket": "/spire-server/api.sock"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.metrics      metrics server is starting to listen    {"addr": "127.0.0.1:8082"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.builder      skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called    {"GVK": "spire.spiffe.io/v1alpha1, Kind=ClusterFederatedTrustDomain"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.builder      Registering a validating webhook        {"GVK": "spire.spiffe.io/v1alpha1, Kind=ClusterFederatedTrustDomain", "path": "/validate-spire-spiffe-io-v1alpha1-clusterfederatedtrustdomain"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/validate-spire-spiffe-io-v1alpha1-clusterfederatedtrustdomain"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.builder      skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called    {"GVK": "spire.spiffe.io/v1alpha1, Kind=ClusterSPIFFEID"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.builder      Registering a validating webhook        {"GVK": "spire.spiffe.io/v1alpha1, Kind=ClusterSPIFFEID", "path": "/validate-spire-spiffe-io-v1alpha1-clusterspiffeid"}
2022-04-13T01:51:18.588Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/validate-spire-spiffe-io-v1alpha1-clusterspiffeid"}
2022-04-13T01:51:18.588Z        INFO    setup   starting manager
I0413 01:51:18.589381      71 leaderelection.go:248] attempting to acquire leader lease spire-system/98c9c988.spiffe.io...
2022-04-13T01:51:18.589Z        INFO    starting metrics server {"path": "/metrics"}
2022-04-13T01:51:18.589Z        INFO    controller-runtime.webhook.webhooks     starting webhook server
2022-04-13T01:51:18.589Z        ERROR   error received after stop sequence was engaged  {"error": "context canceled"}
E0413 01:51:18.589993      71 leaderelection.go:330] error retrieving resource lock spire-system/98c9c988.spiffe.io: Get "https://10.96.0.1:443/api/v1/namespaces/spire-system/configmaps/98c9c988.spiffe.io": context canceled
2022-04-13T01:51:18.590Z        ERROR   setup   problem running manager {"error": "open /tmp/k8s-webhook-server/serving-certs/tls.crt: no such file or directory"}
```

I have tried using Kind v0.11.1 and the latest v0.12.0 on Ubuntu 20.04

**Comment by azdagron** [comment]
Unfortunately, the demo uses the nightly build and the webhook functionality has not been completed. I'll dedicate some time today to wrap that up and update the demo.

**Comment by azdagron** [comment]
I made good progress this week and have things working again with the webhook. I have one more small bit of implementation to add that I hope to get done monday.

**Comment by azdagron** [comment]
I've merged a PR (#13) that completes the automatic webhook certificate management and have updated the demo accordingly. Please let me know if you still run into problems! Sorry it took so long to get this addressed.
